src/hierarchies/nnig_hierarchy.hpp

```diff
@@ -37,18 +37,39 @@ class NNIGHierarchy : public BaseHierarchy {
   };
 
  protected:
+  double data_sum = 0;
+  double data_sum_squares = 0;
   // STATE
   State state;
   // HYPERPARAMETERS
   std::shared_ptr<Hyperparams> hypers;
   // HYPERPRIOR
   std::shared_ptr<bayesmix::NNIGPrior> prior;
 
+  void reset() {
+    data_sum = 0;
+    data_sum_squares = 0;
+    card = 0;
+    cluster_data = std::unordered_map<int, Eigen::VectorXd>();
+  }
```

Comment:
This function should have a better name. `clear_data()` comes to mind, for example